modules/src/core/ics26_routing/handler.rs

```diff
@@ -46,13 +53,16 @@ pub fn decode(message: Any) -> Result<Ics26Envelope, Error> {
 /// Top-level ICS dispatch function. Routes incoming IBC messages to their corresponding module.
 /// Returns a handler output with empty result of type `HandlerOutput<()>` which contains the log
 /// and events produced after processing the input `msg`.
+/// If this method returns an error, the runtime is expected to rollback all state modifications to
+/// the `Ctx` caused by all messages from the transaction that this `msg` is a part of.
 pub fn dispatch<Ctx>(ctx: &mut Ctx, msg: Ics26Envelope) -> Result<HandlerOutput<()>, Error>
```

Comment:
Wondering if it would be better to change the `ics26_routing::handler::deliver()` function to work with a single `message: Any`. It is possible that a tx has multiple messages of which some might not be ibc messages, this makes it the current `deliver()` implementation difficult to use.